src/mutation/transactPutItems.ts

```diff
@@ -12,15 +13,25 @@ export function transactPutItems(
   dbClient: DocumentClient,
   table: TableConfig,
   items: Array<AnyObject>,
-): Promise<PromiseResult<DocumentClient.TransactWriteItemsOutput, AWSError>> {
-  return dbClient
-    .transactWrite({
-      TransactItems: items.map<TransactWriteItem>(x => ({
-        Put: {
-          TableName: table.name,
-          Item: x,
-        },
-      })),
-    })
-    .promise();
+): Promise<
+  Array<PromiseResult<DocumentClient.TransactWriteItemsOutput, AWSError>>
+> {
+  // TransactWriteItems accepts maximum of 100 items, 4 MB total and 400 KB per each item
+  // https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_TransactWriteItems.html
+  const batches = chunk(items, 100) as Array<AnyObject[]>;
+
+  return Promise.all(
+    batches.map(x =>
+      dbClient
+        .transactWrite({
+          TransactItems: x.map<TransactWriteItem>(x => ({
+            Put: {
+              TableName: table.name,
+              Item: x,
+            },
+          })),
+        })
+        .promise(),
+    ),
+  );
```

Comment:
This cannot be done. Transact write is limited to 100 items at once. It is a very expensive operation and if we need to use transact write for more than 10 records at once it will indicate a system design issue.

if you split TransactWrite operation into multiple calls of 100 items how can DynamoDB ensure that the items are inserted in all or nothing way?

This will invalidate the main point of using TransactWrite operation.